helm/dendrite/templates/ingress.yaml

```diff
@@ -1,28 +1,36 @@
 {{- if .Values.ingress.enabled -}}
-{{ $component := "monolith" }}
-{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- $fullName := include "dendrite.fullname" . -}}
+  {{- $svcPort := .Values.service.port -}}
+  {{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
   {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
   {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
   {{- end }}
-{{- end }}
-{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+  {{- end }}
+  {{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1
-{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+  {{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
+  {{- else -}}
 apiVersion: extensions/v1beta1
-{{- end }}
+  {{- end }}
 kind: Ingress
 metadata:
-  namespace: {{ $.Release.Namespace }}
-  name: {{ $component }}
+  name: {{ $fullName }}
   labels:
-    app: {{ $.Chart.Name }}
-    component: {{ $component }}
-  {{- with .Values.ingress.annotations }}
+    {{- include "dendrite.labels" . | nindent 4 }}
   annotations:
+    {{- if eq .Values.ingress.annotateFor "nginx" }}
+    nginx.ingress.kubernetes.io/upstream-hash-by: "$http_authorization"
```

Comment:
Realistically we don't support this in Dendrite I think? I'd remove `annotateFor` entirely.